lib/search/best_bets_checker.rb

```diff
@@ -109,23 +88,20 @@ def select_bets(bets)
     def fetch_bets
       analyzed_users_query = " #{@index.analyzed_best_bet_query(@query)} "
       es_response = @index.raw_search(lookup_payload, "best_bet")
-      result = []
+
       es_response["hits"]["hits"].map do |hit|
         details = JSON.parse(Array(hit["fields"]["details"]).first)
-        bet_query, _, bet_type = hit["_id"].rpartition('-')
+        _bet_query, _, bet_type = hit["_id"].rpartition('-')
         stemmed_query_as_term = Array(hit["fields"]["stemmed_query_as_term"]).first
 
         # The search on the stemmed_query field is overly broad, so here we need
         # to filter out such matches where the query in the bet is not a
         # substring (modulo stemming) of the user's query.
-        unless stemmed_query_as_term.nil?
-          unless analyzed_users_query.include? stemmed_query_as_term
-            next
-          end
-        end
-        result << [bet_query, bet_type, details["best_bets"], details["worst_bets"]]
+        next if stemmed_query_as_term && !analyzed_users_query.include?(stemmed_query_as_term)
```

Comment:
Might just be me, but I think the use of `compact` isn't obvious here. I'd prefer the `nil` case to be more explicit, like

```ruby
if stemmed_query_as_term && ....
  nil
else
  [bet_type, details["best_bets"], details["worst_bets"]]
end
```